torch/utils/cpp_backtrace.py

```diff
@@ -9,3 +12,19 @@ def get_cpp_backtrace(frames_to_skip=0, maximum_number_of_frames=64) -> str:
         maximum_number_of_frames (int): the maximum number of frames to return
     """
     return _get_cpp_backtrace(frames_to_skip, maximum_number_of_frames)
+
+def get_python_cpp_trace():
+    from torch._C._profiler import gather_traceback, symbolize_tracebacks
+    tb = symbolize_tracebacks([gather_traceback(python=True, script=True, cpp=True)])[0]
```

Comment:
We already have a function on the c++ side that does this collection and cleanup, should we just re-use it instead of re-implementing the logic here?

Reply:
good question. I will use Ed's `CapturedTraceback` as you suggested. a side note is `_get_cpp_backtrace` use on linux's `backtrace` somehow it always segfault on my devgpu.  `CapturedTraceback` use `unwind` and it works well for me so far